src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx

```diff
@@ -0,0 +1,104 @@
+import { Localized } from "@fluent/react/compat";
+import React, { FunctionComponent, useMemo } from "react";
+import { graphql } from "react-relay";
+
+import { getModerationLink } from "coral-framework/helpers";
+import { useLocal, withFragmentContainer } from "coral-framework/lib/relay";
+import { Ability, can } from "coral-framework/permissions";
+import { Button } from "coral-ui/components/v3/Button/Button";
+
+import { ModerateStoryButton_local } from "coral-admin/__generated__/ModerateStoryButton_local.graphql";
+import { ModerateStoryButton_settings } from "coral-admin/__generated__/ModerateStoryButton_settings.graphql";
+import { ModerateStoryButton_story } from "coral-admin/__generated__/ModerateStoryButton_story.graphql";
+import { ModerateStoryButton_viewer } from "coral-admin/__generated__/ModerateStoryButton_viewer.graphql";
+
+import styles from "./ModerateStoryButton.css";
+
+export interface Props {
+  settings: ModerateStoryButton_settings;
+  viewer: ModerateStoryButton_viewer;
+  story: ModerateStoryButton_story;
+}
+
+const ModerateStoryButton: FunctionComponent<Props> = ({
+  settings,
+  story: { id, isArchived, isArchiving, canModerate },
+  viewer,
+}) => {
+  const [{ accessToken }] = useLocal<ModerateStoryButton_local>(graphql`
+    fragment ModerateStoryButton_local on Local {
+      accessToken
+    }
+  `);
+
+  const href = useMemo(() => {
+    // TODO: (marcushaddon) this weil; need to be updated once shadow dom is merged!
+    let link = getModerationLink({ storyID: id });
+    if (
+      accessToken &&
+      settings.auth.integrations.sso.enabled &&
+      settings.auth.integrations.sso.targetFilter.admin
+    ) {
+      link += `#accessToken=${accessToken}`;
+    }
+
+    return link;
+  }, [accessToken, settings, id]);
```

Comment:
We have this pattern of using `getModerationLink` and then doing the rigamarole of appending an `accessToken` to it.

How do we feel about updating the `getModerationLink` to take an optional `accessToken` parameter so that we can append in this access token more cleanly in one line and not have duplicates of this append logic all over the admin?

Reply:
I like that idea. That function does change to a hook in the Shadow DOM PR, and I have a note to update this PR/area of the code once that is merged, maybe it would be easier to do it then?